core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -527,6 +528,10 @@ public void onFileInit(Path file) {
             if (logger.isTraceEnabled()) {
                 logger.trace("Loading script file : [{}]", file);
             }
+            if (FileSystemUtils.isHidden(file)) {
```

Comment:
I think this should go before the previous check because otherwise a trace log will read that the script file is being loaded, and then that it is being skipped.